torch/csrc/jit/passes/prepack_folding.cpp

```diff
@@ -0,0 +1,59 @@
+#include <stack>
+
+#include <torch/csrc/jit/api/module.h>
+#include <torch/csrc/jit/passes/constant_pooling.h>
+#include <torch/csrc/jit/passes/constant_propagation.h>
+#include <torch/csrc/jit/passes/prepack_folding.h>
+
+namespace torch {
+namespace jit {
+
+// Must run this pass after constant folding.
+void FoldPrePackingOps(script::Module& m,
+    const PrePackingOpsFilterFn& is_foldable_op) {
+  // Since this pass can be called by quantization or other passes as well,
+  // we need to make sure we generate a unique "packed_weight_\d+"
+  // Thus static uid.
+  static int64_t uid = 0;
```

Comment:
This makes me nervous.  It's not thread safe, and running the pass multiple times will change the results, which can be confusing.  Is there some way to keep track of this in the module that's being transformed?

Reply:
Thanks for the input David. I missed the thread-safety of this. I think we can borrow module name. Note that the intention here is that this pass will be run after freezing, which does inlining as well. So given that module hierarchy flattened by the time we come here, we can assume that only top level module exists and only for that reason this approach works. 
For now this pass is exposed as a JIT pass on python side for testing independently. But we probably want to hide this from JIT python API and wrap it as a part of "XNNPACKify...". What do you think? @dreiss

Reply:
Yeah, I think we'll want to wrap it in a higher-level function.

But I still think that 

```
A = optimize(torch.jit.load(foo))
B = optimize(torch.jit.load(foo))
```

should produce identical A and B.  With the current code, I don't believe that is the case.

Reply:
We can discuss this, but the two passes, 1) convert linear/convs with their packed variants and 2) remove prepacking ops, should pass the test above, especially after fixing this PR for the issue you brought up.
Additionally it should also give us the right answer if we do:
```
A = optimize(torch.jit.load(foo))
B = optimize(A)
```
But you might have caught something I missed.